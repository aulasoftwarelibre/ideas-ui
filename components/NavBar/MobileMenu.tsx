import React from "react";
import GradientButton from "../GradientButton";

export const MobileMenu: React.FunctionComponent = () => {
  return (
    <div className="absolute top-0 inset-x-0 p-2 transition transform origin-top-right md:hidden">
      <div className="rounded-lg shadow-lg">
        <div className="rounded-lg shadow-xs bg-white divide-y-2 divide-gray-50">
          <div className="pt-5 pb-6 px-5 space-y-6">
            <div className="flex items-center justify-between">
              <div>
                <img
                  className="h-8 w-auto"
                  src="/logotipo-completo-degradado.svg"
                  alt="Workflow"
                />
              </div>
              <div className="-mr-2">
                <button
                  type="button"
                  className="inline-flex items-center justify-center p-2 rounded-md text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 focus:text-gray-500 transition duration-150 ease-in-out"
                >
                  <svg
                    className="h-6 w-6"
                    fill="none"
                    viewBox="0 0 24 24"
                    stroke="currentColor"
                  >
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2"
                      d="M6 18L18 6M6 6l12 12"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div>
              <nav className="grid row-gap-8">
                <a
                  href="#"
                  className="-m-3 p-3 flex items-center space-x-3 rounded-md hover:bg-gray-50 transition ease-in-out duration-150"
                >
                  <div className="text-base leading-6 font-medium text-gray-900">
                    Inicio
                  </div>
                </a>
                <a
                  href="#"
                  className="-m-3 p-3 flex items-center space-x-3 rounded-md hover:bg-gray-50 transition ease-in-out duration-150"
                >
                  <div className="text-base leading-6 font-medium text-gray-900">
                    Organizaciones
                  </div>
                </a>
                <a
                  href="#"
                  className="-m-3 p-3 flex items-center space-x-3 rounded-md hover:bg-gray-50 transition ease-in-out duration-150"
                >
                  <div className="text-base leading-6 font-medium text-gray-900">
                    Ayuda
                  </div>
                </a>
              </nav>
            </div>
          </div>
          <div className="py-6 px-5 space-y-6">
            <div className="space-y-6">
              <GradientButton>Iniciar sesi√≥n</GradientButton>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default MobileMenu;
