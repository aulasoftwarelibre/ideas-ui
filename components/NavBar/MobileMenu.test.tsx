import React from 'react';

import { renderMockRouter, screen } from '../../utils/test-utils';
import { Props } from './MobileMenu';
import { Hidden, WithActiveLink, WithoutActiveLink } from './MobileMenu.stories';

describe("MobileMenu", () => {
  it("should be rendered without active link", () => {
    renderMockRouter(
      <WithoutActiveLink {...(WithoutActiveLink.args as Props)} />,
      {
        router: { pathname: "/no-active" },
      }
    );

    expect(screen.getByTestId("mobile-menu"));
    expect(screen.queryAllByTestId("active-mobile-link")).toHaveLength(0);
  });

  it("should be rendered with active link", () => {
    renderMockRouter(<WithActiveLink {...(WithActiveLink.args as Props)} />, {
      router: { pathname: "/" },
    });

    expect(screen.getByTestId("mobile-menu"));
    expect(screen.queryAllByTestId("active-mobile-link")).toHaveLength(1);
  });

  it("should be hidden when menu is closed", () => {
    renderMockRouter(<Hidden {...(Hidden.args as Props)} />, {
      router: { pathname: "/" },
    });

    expect(screen.getByTestId("mobile-menu")).toHaveClass("sm:hidden");
  });
});
