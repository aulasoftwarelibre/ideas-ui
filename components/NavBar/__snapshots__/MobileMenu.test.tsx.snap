// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu should be rendered with active link 1`] = `
<DocumentFragment>
  <div
    class="sm:hidden block"
  >
    <div
      class="px-2 pt-2 pb-3 bg-white"
    >
      <span
        class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-white bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
      >
        Inicio
      </span>
      <a
        class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        href="/groups"
      >
        Grupos
      </a>
      <a
        class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        href="/ideas/new"
      >
        Añadir idea
      </a>
      <a
        class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        href="/help"
      >
        Ayuda
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MobileMenu should be rendered without active link 1`] = `
<DocumentFragment>
  <div
    class="sm:hidden block"
  >
    <div
      class="px-2 pt-2 pb-3 bg-white"
    >
      <a
        class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        href="/"
      >
        Inicio
      </a>
      <a
        class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        href="/groups"
      >
        Grupos
      </a>
      <a
        class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        href="/ideas/new"
      >
        Añadir idea
      </a>
      <a
        class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        href="/help"
      >
        Ayuda
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MobileMenu should not be rendered when closed 1`] = `
<DocumentFragment>
  <div
    class="sm:hidden hidden"
  >
    <div
      class="px-2 pt-2 pb-3 bg-white"
    >
      <span
        class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-white bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
      >
        Inicio
      </span>
      <a
        class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        href="/groups"
      >
        Grupos
      </a>
      <a
        class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        href="/ideas/new"
      >
        Añadir idea
      </a>
      <a
        class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        href="/help"
      >
        Ayuda
      </a>
    </div>
  </div>
</DocumentFragment>
`;
